### PR TITLE
Add section on MPI

### PR DESCRIPTION
**Pull request by awadell1**

### CODE REVIEW

**Comment by awadell1** [comment]
@mkphuthi - Thoughts?

**Comment by abillscmu** [comment]
@mkphuthi please review

**Review by abillscmu — APPROVED** [review]

**Comment by abillscmu** [comment]
@awadell1 please review

File: about/user_software.md [hunk 1]
```diff
@@ -36,13 +40,47 @@ When spack packages are installed, they are accompanied by a modulefile. Therefo
...
+Arjuna provides MPI support via the [OpenMPI](https://www.open-mpi.org/) module (`module load openmpi/4.1.3`).
+Users should load this module if they are using MPI. The following prior installations are deprecated and not reccomended:
+
+- `module load openmpi/4.1.1`
+- MPI Installation at `/usr/local/mpirun`
+
+> `/usr/local/mpirun` is on the `PATH` by default. However it is **not supported**, and will be removed in the future.
+> Users are __strongly__ encouraged to use `module load openmpi/4.1.3` instead.
+
+Using legacy launchers (i.e. `mpirun` or `mpiexec`) is __not supported__ using `openmpi/4.1.3` and __not recommended__ on other mpi installations. Please use `srun` to launch mpi jobs.
```
Comment (awadell1) [inline about/user_software.md:62]: What? Should this be a list?

File: about/user_software.md [hunk 2]
```diff
@@ -36,13 +40,47 @@ When spack packages are installed, they are accompanied by a modulefile. Therefo
...
+
+- `module load openmpi/4.1.1`
+- MPI Installation at `/usr/local/mpirun`
+
+> `/usr/local/mpirun` is on the `PATH` by default. However it is **not supported**, and will be removed in the future.
+> Users are __strongly__ encouraged to use `module load openmpi/4.1.3` instead.
+
+Using legacy launchers (i.e. `mpirun` or `mpiexec`) is __not supported__ using `openmpi/4.1.3` and __not recommended__ on other mpi installations. Please use `srun` to launch mpi jobs. 
+
+If users need a different version of mpi than the ones provided, they can install their own via spack or build their own from source. Users are advised to first read the [slurm documentation regarding mpi](https://slurm.schedmd.com/mpi_guide.html) before installing mpi, and make sure to build mpi with pmi and slurm support.
```
Comment (awadell1) [inline about/user_software.md:64]: "If a user needs a version of MPI other than the ones provides, they may install their own using `spack` or other build tools. "

Capitalize "MPI" and "PMI"

File: about/user_software.md [hunk 3]
```diff
@@ -36,13 +40,47 @@ When spack packages are installed, they are accompanied by a modulefile. Therefo
...
 ```
 
-## Additional Software
 
-In general, we recommend users install additional software to their home directory
-and independently manage/maintain it. 
+## MPI
+> This section will discuss usage of MPI on Arjuna, and presumes familiarity with MPI.
+
+Arjuna provides MPI support via the [OpenMPI](https://www.open-mpi.org/) module (`module load openmpi/4.1.3`).
```
Comment (awadell1) [inline about/user_software.md:53]: Change to `module load openmpi` 
I will set the default to be `openmpi/4.1.3`

File: about/user_software.md [hunk 4]
```diff
@@ -36,13 +40,47 @@ When spack packages are installed, they are accompanied by a modulefile. Therefo
...
+
+> `/usr/local/mpirun` is on the `PATH` by default. However it is **not supported**, and will be removed in the future.
+> Users are __strongly__ encouraged to use `module load openmpi/4.1.3` instead.
+
+Using legacy launchers (i.e. `mpirun` or `mpiexec`) is __not supported__ using `openmpi/4.1.3` and __not recommended__ on other mpi installations. Please use `srun` to launch mpi jobs. 
+
+If users need a different version of mpi than the ones provided, they can install their own via spack or build their own from source. Users are advised to first read the [slurm documentation regarding mpi](https://slurm.schedmd.com/mpi_guide.html) before installing mpi, and make sure to build mpi with pmi and slurm support.
+
+### PMI
+Arjuna has support for `pmi`, `pmi2`, and `pmix`. To run a job using a specified pmi, launch your jobs using the `--mpi` flag provided by `srun`.
```
Comment (awadell1) [inline about/user_software.md:67]: If you are unsure what to use, we suggest trying --mpi=pmix first”

File: about/user_software.md [hunk 5]
```diff
@@ -36,13 +40,47 @@ When spack packages are installed, they are accompanied by a modulefile. Therefo
...
 
-## Additional Software
 
-In general, we recommend users install additional software to their home directory
-and independently manage/maintain it. 
+## MPI
+> This section will discuss usage of MPI on Arjuna, and presumes familiarity with MPI.
+
+Arjuna provides MPI support via the [OpenMPI](https://www.open-mpi.org/) module (`module load openmpi/4.1.3`).
+Users should load this module if they are using MPI. The following prior installations are deprecated and not reccomended:
```
Comment (awadell1) [inline about/user_software.md:54]: recommended

File: about/user_software.md [hunk 6]
```diff
@@ -36,13 +40,47 @@ When spack packages are installed, they are accompanied by a modulefile. Therefo
...
+
+- `module load openmpi/4.1.1`
+- MPI Installation at `/usr/local/mpirun`
+
+> `/usr/local/mpirun` is on the `PATH` by default. However it is **not supported**, and will be removed in the future.
+> Users are __strongly__ encouraged to use `module load openmpi/4.1.3` instead.
```
Comment (awadell1) [inline about/user_software.md:60]: `module load openmpi`

File: about/user_software.md (hunk 1 above)
Comment (abillscmu) [reply about/user_software.md:62]: no, the sentence as written is exhaustive, therefore it should be i.e. the only things it excludes are mpirun and mpiexec, which it refers to as the "legacy launchers"

**Pull request merged**